edb/lib/std/25-numoperators.edgeql

```diff
@@ -1781,8 +1776,24 @@ CREATE CAST FROM std::float32 TO std::bigint {
 
 
 CREATE CAST FROM std::float32 TO std::decimal {
-    SET volatility := 'IMMUTABLE';
-    USING SQL CAST;
+    SET volatility := 'STABLE';
+    USING SQL $$
+        SELECT
+            (CASE WHEN val != 'NaN'
+                       AND val != 'Infinity'
+                       AND val != '-Infinity'
+            THEN
+                val::numeric
+            ELSE
+                edgedb._raise_specific_exception(
+                    'invalid_text_representation',
+                    'invalid value for numeric: ' || quote_literal(val),
```

Comment:
```suggestion
                    'invalid value for decimal: ' || quote_literal(val),
```
?

Reply:
This gets replaced by `errormech` error normalization.